packages/plugin-npm/sources/npmHttpUtils.ts

```diff
@@ -84,17 +84,19 @@ export async function post(path: string, body: httpUtils.Body, {attemptedAs, con
   const auth = await getAuthenticationHeader(registry, {authType, configuration, ident});
   if (auth)
     headers = {...headers, authorization: auth};
+  if (otp)
+    headers = {...headers, ...getOtpHeaders(otp)}
 
   try {
     return await httpUtils.post(registry + path, body, {configuration, headers, ...rest});
   } catch (error) {
-    if (!isOtpError(error)) {
+    if (!isOtpError(error) || otp) {
```

Comment:
IMHO, if the `otp` has been provided explicitly then you should not try to ask for it if it turns out to be invalid - let the "caller" to deal with this.

Reply:
Seems reasonable 👍